frontend/src/modules/search/components/FiltersPanel.js

```diff
@@ -33,23 +36,42 @@ type State = {|
 export class FiltersPanelBase extends React.Component<Props, State> {
     constructor(props: Props) {
         super(props);
+
         this.state = {
             visible: false,
         };
     }
 
     toggleVisibility = () => {
-        this.setState((state) => {
+        this.setState(state => {
             return { visible: !state.visible };
         });
     }
 
-    selectStatus(status: string) {
-        const newStatus = status === 'all' ? null : status;
+    applyFilters = () => {
+        this.toggleVisibility();
+        return this.props.update();
+    }
+
+    resetFilters = () => {
+        this.props.resetStatuses();
+    }
+
+    createHandleSelectStatus = (status: string) => {
+        if (status === 'all') {
+            return null;
+        }
+
+        return (event: SyntheticInputEvent<>) => {
+            event.stopPropagation();
+            this.props.toggleStatus(status);
+        };
+    }
 
+    createSetStatus(status: string) {
```

Comment:
I believe we need an exception here in case of `All`. Right now clicking on `All` threats it as a filter: when you reopen the filter list, it says "Apply 1 filter" and when you tick "Translated", the count increases and the placeholder reads "All, Translated".